### PR TITLE
Chitter challenge submission (Ara)

### PR DESCRIPTION
**Pull request by aracho1**
-Refer to README 
-Completed all user stories including tagging users and sending emails
-I tried using Data Mappers and host the app on Heroku but there were configuration issues

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,6 +5,11 @@ services:
```
Comment (lukewickens1989) [inline .travis.yml:5]: love the detailed instructions in your README. It's given me some great tips on how to improve mine thanks Ara :)

File: README.md [hunk 2]
```diff
@@ -45,91 +48,21 @@ As a Maker
 So that I can avoid others posting messages on Chitter as me
 I want to log out of Chitter
 
-ADVANCED
 
 As a Maker
 So that I can stay constantly tapped in to the shouty box of Chitter
 I want to receive an email if I am tagged in a Peep
 ```
+Screenshots:
```
Comment (lukewickens1989) [inline README.md:56]: great screenshots!

File: app/app.rb [hunk 3]
```diff
@@ -0,0 +1,118 @@
+ENV['ENVIRONMENT'] ||= 'development'
+
+require 'sinatra/base'
```
Comment (lukewickens1989) [inline app/app.rb:3]: i like that you have split the models up

File: app/app.rb [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+require 'mail'
+
+
+class Chitter < Sinatra::Base
+  enable :sessions, :method_override
+  set :session_secret, 'super secret'
+  register Sinatra::Flash
+
+  get '/' do
+    erb :login, :layout => :layout
```
Comment (lukewickens1989) [inline app/app.rb:20]: I havent seen this before I will need to have a look into what it does.

File: app/app.rb [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+      flash[:notice] = "Welcome back, #{user.username}!"
+      redirect '/home'
+    else
+      flash[:notice] = "The email or password did not match our records. Please try again."
+      redirect '/'
+    end
+  end
+
+  get '/follow/:username' do
+    flash[:notice] = "You are now following @#{params[:username]}"
```
Comment (lukewickens1989) [inline app/app.rb:57]: I like the use of all the flash notices to the user to let them know what theyve done and how the process went

File: lib/seeds.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+require './app/models/message'
```
Comment (lukewickens1989) [inline lib/seeds.rb:1]: I like the use of the seeds file to reduce the text used in the tests/development

**Review by cryptonoyaiba — COMMENTED** [review]
The app works great and it looks like you have succeeded in every user story. The styling of the app is great and actually looks like something you would find being released by a company. I like how you have structured your files and split away functionality to further separate concerns. Some of the code I have no idea what it does, so you have given me quite a bit to look over . I thought your README file was great and really helpful in understanding the product and seeing how it worked visually. I only had 2 issues when running RSpec (which maybe something I have done in setting up the database on my end.)
`
posting a message app does not throw an error when special characters are used
     Failure/Error: fill_in 'new_message', with:"Let's sing!"
     
     Capybara::ElementNotFound:
       Unable to find field "new_message" that is not disabled

Sign out user can sign out
     Failure/Error: click_on 'Sign Out'
     
     Capybara::ElementNotFound:
       Unable to find link or button "Sign Out"

`
The only thing I can think of as an addition to what you have is some preloaded comments in order to view the cheeps page and see some content there already.
Great job Ara!

**Pull request closed without merging**